src/test/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactoryTest.java

```diff
@@ -0,0 +1,147 @@
+		assertEquals(floorAreaTerrace, apart.getFloorAreaTerrace());
+		assertEquals(description, apart.getDescription());
+		assertEquals(title, apart.getTitle());
+		assertEquals(wifi, apart.getWifi());
+		assertEquals(pricePerNight, apart.getPricePerNight());
+		assertEquals(nbMinNight, apart.getNbMinNight());
+		assertEquals(tele, apart.getTele());
+	}
+
+	/**
+	 * this function tests the immutability of an instance of Apartment using
+	 * Method invoke to check the accessibility of setters.
+	 */
+	@Test
+	void testImmutableApartment() {		
+		Apartment apart = ApartmentFactory.generateRandomApartment();
+		//Test boolean Setters Permissions
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setWifi", boolean.class).invoke(apart, true);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setTele", boolean.class).invoke(apart, true);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setTerrace", boolean.class).invoke(apart, true);  });
+
+		//Test double Setters Permissions
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setFloorArea", double.class).invoke(apart, 5.5);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setFloorAreaTerrace", double.class).invoke(apart, 5.5);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setPricePerNight", double.class).invoke(apart, 5.5);  });
+
+		//Test String Setters Permissions
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setAddress", String.class).invoke(apart, "Don't");  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setDescription", String.class).invoke(apart, "Do");  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setTitle", String.class).invoke(apart, "That ! :o");  });
+
+		//Test int Setters Permissions
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setNbBedrooms", int.class).invoke(apart, 5);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setNbSleeping", int.class).invoke(apart, 5);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setNbBathrooms", int.class).invoke(apart, 5);  });
+		assertThrows(NoSuchMethodException.class,
+				()->{apart.getClass().getMethod("setNbMinNight", int.class).invoke(apart, 5);  });
+	}
+
+	/**
+	 * This class tests the validity of randomly generated apartment
+	 * by the function generateRandomApartment() used by generateRandomApartmentList()
+	 * The test are done with 100 apartments in order to have a representative list of Apartments.
+	 * It will allow us to have a huge range of values and check the validity of all parameters.
+	 */
+	@Test
+	void TestGenerateRandomAparmentsList() {
```

Comment:
Commencer par une minuscule SVP.

Reply:
Resolved in commit b21abff4a2e191eb00dedc4959d7668b9b20cd8b